### DISCUSSION TITLE
UnhandledSchemeError: Reading from "node:stream"

### DISCUSSION THREAD

**Discussion by kousherAlam** [webpack/webpack#18740] [General]
I'm working on a nextjs project, in the middleware file i used firebase-admin package which using node:stream package internally. But i'm getting this error, how can i resolve this ?

```
Module build failed: UnhandledSchemeError: Reading from "node:stream" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "node:" URIs
```

![Screenshot from 2024-09-04 21-47-48](https://github.com/user-attachments/assets/a8e07e1e-1838-408d-8f76-8d7921d2d33a)

package version
```
"firebase-admin": "^12.4.0",
"next": "^14.2.7",
```

**Comment by alexander-akait** [comment]
You can't bundle `node:stream` for `web` target, you need Polyfill them using `externals`

**Comment by kousherAlam** [comment]
@alexander-akait thanks for your response bro, but still getting the error after update config. here is the minimum reproduceable project. 
https://stackblitz.com/edit/stackblitz-starters-twdrzj?file=middleware.ts

**Reply by Faraz643** [reply]
did you resolve the issue? I am also facing the same

**Reply by kousherAlam** [reply]
no bro i still has the issue.... please let me know if you find soln

**Reply by kousherAlam** [reply]
@Faraz643 https://discord.com/channels/752553802359505017/1007476603422527558/threads/1276311253563871272

**Reply by alexander-akait** [reply]
Sorry for delay, but `node:stream` in not avaliable in this context - https://nextjs.org/docs/app/building-your-application/routing/middleware#runtime and https://nextjs.org/docs/app/api-reference/edge

We can't fix it here, sorry. I recommend to use `fetch` and browsers streams
